### PR TITLE
Map Gestures - Improve transmit process

### PR DESCRIPTION
**Pull request by jokoho48**
**When merged this pull request will:**
- simplify map gestures transmit process
- Reduces network traffic
- increases performance
- and opens doors for Zeus and Spectator implementations

### CODE REVIEW

**Comment by jonpas** [comment]
Please rename PR title to match the guidelines as seen in the PR template.

File: addons/map_gestures/functions/fnc_transmit.sqf [hunk 1]
```diff
@@ -28,15 +28,9 @@ if (!GVAR(EnableTransmit) || !visibleMap) exitWith {
...
-    private _owner = _x getVariable [QEGVAR(common,playerOwner), -1];
-    if (_owner > -1) then {
-        private _remotePos = _x getVariable [QGVAR(remotePos), [0,0,0]];
-        if ((_remotePos distance2d GVAR(pointPosition)) > 1) then { // Only transmit when actually moving
-            [QGVAR(syncPos), [ACE_Player, GVAR(pointPosition)], _owner] call CBA_fnc_ownerEvent;
-            _x setVariable [QGVAR(remotePos), GVAR(pointPosition)];
-        };
-    };
-} count ([ACE_player, GVAR(maxRange)] call FUNC(getProximityPlayers));
+private _remotePos = ace_player getVariable [QGVAR(pointPosition), [0,0,0]];
```
Comment (mharis001) [inline addons/map_gestures/functions/fnc_transmit.sqf:31]:
```suggestion
private _remotePos = ACE_player getVariable [QGVAR(pointPosition), [0, 0, 0]];
```

File: addons/map_gestures/functions/fnc_transmit.sqf [hunk 2]
```diff
@@ -28,15 +28,9 @@ if (!GVAR(EnableTransmit) || !visibleMap) exitWith {
...
+if ((_remotePos distance2d GVAR(pointPosition)) > 1) then { // Only transmit when actually moving
```
Comment (mharis001) [inline addons/map_gestures/functions/fnc_transmit.sqf:32]:
```suggestion
if (_remotePos distance2D GVAR(pointPosition) > 1) then { // Only transmit when actually moving
```

File: addons/map_gestures/functions/fnc_transmit.sqf [hunk 3]
```diff
@@ -28,15 +28,9 @@ if (!GVAR(EnableTransmit) || !visibleMap) exitWith {
...
+    ace_player setVariable [QGVAR(pointPosition), GVAR(pointPosition), true];
```
Comment (mharis001) [inline addons/map_gestures/functions/fnc_transmit.sqf:33]:
```suggestion
    ACE_player setVariable [QGVAR(pointPosition), GVAR(pointPosition), true];
```

**Comment by PabstMirror** [comment]
remove `[] call EFUNC(common,setPlayerOwner);` in postInit, I think nothing else uses it now so it won't have to run

File: addons/map_gestures/XEH_postInit.sqf [hunk 4]
```diff
@@ -8,16 +8,11 @@ if (!hasInterface) exitWith {};
 
 ["ace_settingsInitialized", {
     if (!GVAR(enabled)) exitWith {};
-    
+
     // This will set QEGVAR(common,playerOwner) var on player objects
```
Comment (PabstMirror) [inline addons/map_gestures/XEH_postInit.sqf:12]:
```suggestion
```

File: addons/map_gestures/XEH_postInit.sqf [hunk 5]
```diff
@@ -8,16 +8,11 @@ if (!hasInterface) exitWith {};
...
     [] call EFUNC(common,setPlayerOwner);
```
Comment (PabstMirror) [inline addons/map_gestures/XEH_postInit.sqf:13]:
```suggestion
```

**Comment by PabstMirror** [comment]
Just leaving note here in-case we see any issues with network traffic:
alternate fix would be to use setVar with client array

```
private _clientsToSendTo = [];
{
    private _owner = _x getVariable [QEGVAR(common,playerOwner), -1];
    if (_owner > -1) then {
        private _remotePos = _x getVariable [QGVAR(remotePos), [0,0,0]];
        if ((_remotePos distance2d GVAR(pointPosition)) > 1) then { // Only transmit when actually moving
            _clientsToSendTo pushBack _owner;
            _x setVariable [QGVAR(remotePos), GVAR(pointPosition)];
        };
    };
} count ([ACE_player, GVAR(maxRange)] call FUNC(getProximityPlayers));

ACE_player setVariable [QGVAR(pointPosition), GVAR(pointPosition), _clientsToSendTo];
```

**Comment by jonpas** [comment]
> alternate fix would be to use setVar with client array

Another note: that wouldn't allow Spectators and Zeus to see it on the map, which is @jokoho48's plan.

**Pull request merged**